sympy/ntheory/residue_ntheory.py

```diff
@@ -110,6 +112,8 @@ def primitive_root(p):
     p = as_int(p)
     if p < 1:
         raise ValueError('p is required to be positive')
+    if isprime(p) is False:
+        raise NotImplementedError("Not implemented for composite p")
```

Comment:
Primitive roots exist also for some composite moduli (e.g. odd prime powers) and the implementation below takes care of those cases as well. So these two lines should be removed.